client/my-sites/upgrades/domain-management/transfer/transfer-to-other-site/index.jsx

```diff
@@ -0,0 +1,193 @@
+				closeDialog();
+				page( paths.domainManagementList( this.props.selectedSite.slug ) );
+			}, error => {
+				this.setState( { disableDialogButtons: false } );
+				this.props.errorNotice( error.message || defaultErrorMessage );
+				closeDialog();
+			} );
+	}
+
+	handleDialogClose = () => {
+		if ( ! this.state.disableDialogButtons ) {
+			this.setState( { showConfirmationDialog: false } );
+		}
+	}
+
+	render() {
+		if ( ! this.isDataReady() ) {
+			return <DomainMainPlaceholder goBack={ this.goToEdit } />;
+		}
+
+		const { selectedSite, selectedDomainName } = this.props,
+			{ slug } = selectedSite;
+
+		return (
+			<Main className="transfer-to-other-site">
+				<Header
+					selectedDomainName={ selectedDomainName }
+					backHref={ paths.domainManagementTransfer( slug, selectedDomainName ) }>
+					{ this.props.translate( 'Transfer Domain To Another Site' ) }
+				</Header>
+				{ this.renderSection() }
+			</Main>
+		);
+	}
+
+	renderDialog() {
+		const { selectedDomainName: domainName, translate } = this.props,
+			buttons = [
+				{
+					action: 'cancel',
+					label: translate( 'Cancel' ),
+					disabled: this.state.disableDialogButtons
+				},
+				{
+					action: 'confirm',
+					label: translate( 'Confirm Transfer' ),
+					onClick: this.handleConfirmTransferDomain,
+					disabled: this.state.disableDialogButtons,
+					isPrimary: true
+				}
+			],
+			targetSiteName = get( this.state.targetSite, 'name', '' );
+
+		return (
+			<Dialog isVisible={ this.state.showConfirmationDialog } buttons={ buttons } onClose={ this.handleDialogClose }>
+				<h1>{ translate( 'Confirm Transfer' ) }</h1>
+				<p>{ translate( 'Do you want to transfer {{strong}}%(domainName)s{{/strong}} ' +
+					'to site {{strong}}%(targetSiteName)s{{/strong}}?', {
+						args: { domainName, targetSiteName }, components: { strong: <strong /> }
+					} ) }</p>
+			</Dialog>
+		);
+	}
+
+	renderSection() {
+		const { selectedDomainName: domainName, translate } = this.props,
+			{ currentUserCanManage } = getSelectedDomain( this.props );
+
+		if ( ! currentUserCanManage ) {
+			return <NonOwnerCard { ...omit( this.props, [ 'children' ] ) } />;
+		}
+
+		return (
+			<div>
+				<SectionHeader label={ translate( 'Transfer Domain To Another Site' ) } />
+				<Card className="transfer-to-other-site__card">
+					<p>
+						{ translate( 'Please choose a site you\'re an administrator on to transfer {{strong}}%(domainName)s{{/strong}} to:',
+							{ args: { domainName }, components: { strong: <strong /> } } ) }
+					</p>
+					<SiteSelector
+						filter={ this.checkSiteEligibility }
+						sites={ this.props.sites }
+						onSiteSelect={ this.handleSiteSelect }
+					/>
+				</Card>
+				{ this.renderDialog() }
+			</div>
+		);
+	}
+}
+
+export default connect(
+	state => ( {
+		currentUser: getCurrentUser( state ),
+		sites: getSites( state ),
+		siteBySlug: ( siteSlug ) => getSiteBySlug( state, siteSlug ),
```

Comment:
Thanks for the heads-up! I'll rework this bit and try to avoid the antipattern then 🙇